include/zephyr/settings/settings.h

```diff
@@ -619,6 +619,48 @@ int settings_runtime_commit(const char *name);
  */
 int settings_storage_get(void **storage);
 
+#ifdef CONFIG_SETTINGS_STATIC_DATA
+/**
+ * @struct settings_data_static
+ * Settings data static structure. This can be used to place compile time
+ * data in a format that can be used by the settings subsystem. The settings
+ * static data is registered in a linker section by using a call to
+ * SETTINGS_STATIC_DATA_DEFINE().
+ */
```

Comment:
Suggested update so that the documentation for this actually shows up in the public doxygen documentation.
The `@struct ...` entry is not necessary as Doxygen will just infer that information automatically already.

FWIW it looks like it would be useful to adopt a similar approach for the API that's guarded by `CONFIG_SETTINGS_RUNTIME` so that it shows up as well.

```suggestion
#if defined(CONFIG_SETTINGS_STATIC_DATA) || defined(__DOXYGEN__)
/**
 * Settings data static structure.
 *
 * This can be used to place compile time
 * data in a format that can be used by the settings subsystem. The settings
 * static data is registered in a linker section by using a call to
 * SETTINGS_STATIC_DATA_DEFINE().
 *
 * @kconfig_dep{CONFIG_SETTINGS_STATIC_DATA}
 */
```

Reply:
just (genuinely) curious if you think this too was nitpicking? Surely this can be addressed in a follow-up PR, but on the other hand having proper documentation for public API is kinda important :)

Reply:
Yes I do think this is nitpicking.

A correction like this (that I do not question) belongs in a PR that addresses this in all places where this should be corrected in the header file. When something is added to a file it should follow the same approach as the rest of the file. When a new file is created this should be added right away.

Reply:
But then please at least comment on it, so that I can at least have a chance to remove my -1. I have no way to know that you even agree this needs fixing, nor do I have a way to know you plan on fixing this across the file in the future. If this doesn't get noticed / commented on during reviews like this, then how are existing header files supposed to be fixed?

Reply:
FWIW I should mention that I took a significant amount of time to make sure that you would have a ready-to-apply request for change, which I tested locally, rather than just waving hands about something I would have liked to seen done differently. Yes, fixing this across the entire header would be nice, but in the meanwhile I genuinely thought you'd be happy that this would make the documentation that you effectively took time to write ACTUALLY show up in Doxygen. :/

Reply:
I addressed all items that you required changes for. If afterwards you add another thing how am I to know that the request for changes also implies these new additions.

Thank you for taking the time to have a ready to apply "suggested change", the button "commit suggestion" is tempting but this breaks my workflow so I don't use it (and I don't think anyone should).

Probably I should have said that I think this modification should be done in a follow-up PR, but for me your comment was just a proposal, not a request for changes. Any changes to enhance the documentation should be verified, and this is not a nice process.